app/Services/Account/Settings/ExportAccount.php

```diff
@@ -0,0 +1,1366 @@
+
+class ExportAccount extends BaseService
+{
+    protected $sql;
+
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'user_id' => 'required|integer|exists:users,id',
+        ];
+    }
+
+    /**
+     * Export account as SQL.
+     *
+     * @param array $data
+     * @return string
+     */
+    public function execute(array $data) : string
+    {
+        $this->validate($data);
+
+        $user = User::findOrFail($data['user_id']);
+
+        $this->sql = '# ************************************************************
+# '.$user->first_name.' '.$user->last_name.' dump of data
+# Export date: '.now().'
+# ************************************************************
+
+SET FOREIGN_KEY_CHECKS=0;
+'.PHP_EOL;
+
+        $this->exportAccount($data);
+        $this->exportActivity($data);
+        $this->exportContact($data);
+        $this->exportActivityContact($data);
+        $this->exportActivityStatistic($data);
+        $this->exportActivityTypeCategory($data);
+        $this->exportActivityType($data);
+        $this->exportAddress($data);
+        $this->exportCall($data);
+        $this->exportCompany($data);
+        $this->exportContactFieldType($data);
+        $this->exportContactField($data);
+        $this->exportContactTag($data);
+        $this->exportContact($data);
+        $this->exportConversation($data);
+        $this->exportDays($data);
+        $this->exportDebt($data);
+        $this->exportDocument($data);
+        $this->exportEmotionCall($data);
+        $this->exportEntries($data);
+        $this->exportGender($data);
+        $this->exportGift($data);
+        $this->exportInvitation($data);
+        $this->exportJournalEntry($data);
+        $this->exportLifeEventCategory($data);
+        $this->exportLifeEventType($data);
+        $this->exportLifeEvent($data);
+        $this->exportMessage($data);
+        $this->exportMetaDataLoveRelationship($data);
+        $this->exportModule($data);
+        $this->exportNote($data);
+        $this->exportNotification($data);
+        $this->exportOccupation($data);
+        $this->exportPet($data);
+        $this->exportPhoto($data);
+        $this->exportPlace($data);
+        $this->exportRecoveryCode($data);
+        $this->exportRelationTypeGroup($data);
+        $this->exportRelationType($data);
+        $this->exportRelationship($data);
+        $this->exportReminderOutbox($data);
+        $this->exportReminderRule($data);
+        $this->exportReminderSent($data);
+        $this->exportReminder($data);
+        $this->exportSpecialDate($data);
+        $this->exportTag($data);
+        $this->exportTask($data);
+        $this->exportTermUser($data);
+        $this->exportUser($data);
+        $this->exportWeather($data);
+        $this->exportContactPhoto($data);
```

Comment:
How are we sure this is the right order of recreating(populate) each table?

Reply:
The order doesn't matter as I've disabled the foreign constraints at the beginning of the file. If I remove it, the import is simply impossible due to how we've initially structured the database.